### DISCUSSION TITLE
Infeed threading

### DISCUSSION THREAD

**Discussion by duebelpapst** [kachurovskiy/nanoels#190] [Ideas]
Hello Maxim,

is there a way of modifying the program so that the infeed in X is carried out in such a way that the same clamping cross-section is removed with each infeed?
Or at least so that the infeed decreases by 10% per pass, for example.
I wanted to adapt the code myself, but unfortunately I don't find were and how you programmed this subroutine.

Regards,
Stefan

**Comment by kachurovskiy** [comment]
Hi @duebelpapst, it's defined in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L2777

This line is used for threading, multi-start threading, facing and turning modes.

You could experiment with this by adding something like this in the following code afterwards (untested): 

```
if (mode == MODE_THREAD) {
  float fraction = (turnPasses - ceil(opIndex / float(starts))) / turnPasses;
  fraction = fraction * fraction; // make initial passed larger, final passes smaller
  auxPos = auxEndStop - (auxEndStop - auxStartStop) * fraction;
}
```

**Reply by duebelpapst** [reply]
Hi Maxim,

it works as desired.
Thanks for the quick help.

Regards,
Stefan

**Reply by tektonandroid** [reply]
Hello, Stefan. Exactly where did you put the code that Maxim told you?
Thank you.

**Reply by duebelpapst** [reply]
Hi,

I have inserted the code after line 2777

2777    long auxPos = auxEndStop - (auxEndStop - auxStartStop) / turnPasses * (turnPasses - ceil(opIndex / float(starts)));
2778
2779    if (mode == MODE_THREAD) {
2780      float fraction = (turnPasses - ceil(opIndex / float(starts))) / turnPasses;
2781      fraction = fraction * fraction; // make initial passed larger, final passes smaller
2782      auxPos = auxEndStop - (auxEndStop - auxStartStop) * fraction;
2783    }
2784
2785    // Bringing X to starting position.

Regards,
Stefan

**Reply by tektonandroid** [reply]
Thanks. I will try it.

**Comment by jordanpayn1989** [comment]
I have been toying with the idea of trying to replicate compound advance threading in the software so that only one edge of threading tool is cutting. Thought id check if someone has already done. 

Probably just have to allow a few more encoder pulses pass before Z move. Would this be in the same section @kachurovskiy ?

**Reply by kachurovskiy** [reply]
Hi @jordanpayn1989, sorry for missing this question. To experiment with compound advance I would suggest to play with `spindlePosSync` value we set in https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L2783 - as you can see, currently we implement multi-start threads using it.
